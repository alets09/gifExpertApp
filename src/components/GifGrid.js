import React, { useState , useEffect} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>
            {loading && <p className=" load animate__animated animate__zoomOutDown animate__delay-1s">Loading...</p>}
            <div className="card-grid">
                {
                    data.map(img => (
                        <GifGridItem key={img.id} {...img}/>
                    ))
                }
            </div>
        </> 
    )
}
