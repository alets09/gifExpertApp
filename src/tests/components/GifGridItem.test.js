import React from 'react'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('probar compente de GifGrid', () => {
    test('mostrar el componente correctamente', () => {
        const title = 'titulo';
        const url='https://..';
        const wrapper = shallow( <GifGridItem title={title} url={url}/>);

        expect(wrapper).toMatchSnapshot();
    })
    
})
