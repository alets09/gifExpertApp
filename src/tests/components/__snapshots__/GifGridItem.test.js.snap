// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar compente de GifGrid mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInTopLeft"
>
  <img
    alt="titulo"
    src="https://.."
  />
  <p>
    titulo
  </p>
</div>
`;
